integration-tests/tests/src/tests/sync/open-behavior.ts

```diff
@@ -0,0 +1,658 @@
+    const partitionValue = generatePartition();
+
+    const config: Realm.Configuration = {
+      schema: [DogForSyncSchema],
+      sync: {
+        user,
+        partitionValue,
+        //@ts-expect-error TYPEBUG: cannot access const enum at runtime
+        _sessionStopPolicy: "immediately",
+        //@ts-expect-error TYPEBUG: cannot access const enum at runtime
+        newRealmFileBehavior: { type: "downloadBeforeOpen" },
+      },
+    };
+
+    const openPromise = new Promise((resolve, reject) => {
+      const promise = Realm.open(config);
+      // NOTE: could this potentially trigger before canceling?
+      promise.progress(() => {
+        reject("Progress listener called");
+      });
+      promise.cancel();
+      return promise;
+    });
+
+    openPromise
+      .then(() => {
+        expect(true).to.be.false("Realm was opened after being canceled.");
+      })
+      .catch((err) => {
+        expect(true).to.be.false("An error was thrown after open was canceled: " + err.message);
+      });
+
+    // Wait for 1 second after canceling. The open promise should not emit any events in that period.
+    const timeOutPromise = new Promise<void>((resolve) => {
+      sleep(1000);
+      resolve();
+    });
+
+    const any = Promise.race([timeOutPromise, openPromise]);
+
+    return any.finally(() => user.logOut());
+  });
+  it("opening realm with invalid options throw", async function (this: AppContext) {
+    const user = await this.app.logIn(Realm.Credentials.anonymous());
+    const partitionValue = generatePartition();
+
+    await expect(() =>
+      Realm.open({
+        schema: [DogForSyncSchema],
+        sync: {
+          user,
+          partitionValue,
+          //@ts-expect-error TYPEBUG: cannot access const enum at runtime
+          _sessionStopPolicy: "immediately",
+          //@ts-expect-error testing invalid value for newRealmFileBehaviour
+          newRealmFileBehavior: { type: "foo" },
+        },
+      }),
+    ).throws;
+
+    await expect(() =>
+      Realm.open({
+        schema: [DogForSyncSchema],
+        sync: {
+          user,
+          partitionValue,
+          //@ts-expect-error TYPEBUG: cannot access const enum at runtime
+          _sessionStopPolicy: "immediately",
+          //@ts-expect-error testing invalid value for newRealmFileBehaviour
+          newRealmFileBehavior: { type: "openLocalRealm", timeOutBehavior: "foo" },
+        },
+      }),
+    ).throws;
+
+    await expect(() =>
+      Realm.open({
+        schema: [DogForSyncSchema],
+        sync: {
+          user,
+          partitionValue,
+          //@ts-expect-error TYPEBUG: cannot access const enum at runtime
+          _sessionStopPolicy: "immediately",
+          //@ts-expect-error testing invalid value for newRealmFileBehaviour
+          newRealmFileBehavior: { type: "openLocalRealm", timeOut: "bar" },
+        },
+      }),
+    ).throws;
+    await user.logOut();
```

Comment:
Note for refactor: If the test fails, this will never happen and if other tests are relying on this, it should probably be put into an `afterEach` hook instead.